src/suricata.c

```diff
@@ -2837,6 +2844,14 @@ static void SuricataMainLoop(SCInstance *suri)
             DetectEngineReloadSetIdle();
         }
 
+        if (flush_wait_count) {
+            if (++wait_count == flush_wait_count) {
+                SCLogDebug("Sending flush packet to threads");
+                DetectEngineFlush();
```

Comment:
@victorjulien suggests moving this out of the main thread as the main thread can block for events -- e.g., rule reload.

Reply:
Will create a new thread for this function since the main thread can block (e.g., during rule reloads).